include/odp/api/spec/pool_ext_types.h

```diff
@@ -0,0 +1,221 @@
+
+	/** Maximum number of pools
+	 *
+	 *  Maximum number of external memory pools of the requested type. */
+	uint32_t max_pools;
+
+	/** Minimum size of thread local cache */
+	uint32_t min_cache_size;
+
+	/** Maximum size of thread local cache */
+	uint32_t max_cache_size;
+
+	/** Supported statistics counters */
+	odp_pool_stats_opt_t stats;
+
+	/** Packet pool capabilities  */
+	struct {
+		/** Maximum number of packet buffers */
+		uint32_t max_num_buf;
+
+		/** Maximum packet buffer size in bytes */
+		uint32_t max_buf_size;
+
+		/** ODP header size in bytes
+		 *
+		 *  Application must reserve this many bytes from the start of a packet buffer
+		 *  for ODP implementation usage. When the value is zero, ODP implementation does
+		 *  not need header space to be reserved for it. Application will not modify this
+		 *  memory area (after buffer populate call).
+		 */
+		uint32_t odp_header_size;
+
+		/** ODP trailer size in bytes
+		 *
+		 *  Application must reserve this many bytes from the end of a packet buffer
+		 *  for ODP implementation usage. When the value is zero, ODP implementation does
+		 *  not need trailer space to be reserved for it. Application will not modify this
+		 *  memory area (after buffer populate call).
+		 */
+		uint32_t odp_trailer_size;
+
+		/** Minimum packet buffer pointer alignment in bytes
+		 *
+		 *  Packet buffer pointers populated into the pool must have at least
+		 *  this (power of two) alignment.
+		 */
+		uint32_t min_buf_align;
+
+		/** Minimum packet headroom alignment in bytes
+		 *
+		 *  Packet buffers populated into the pool must have at least this (power of two)
+		 *  alignment for the headroom.
+		 */
+		uint32_t min_head_align;
+
+		/** Packet buffer alignment flags
+		 *
+		 *  These flags specify additional alignment requirements for packet buffers.
+		 *  If not stated otherwise, min_buf_align and min_head_align alignment
+		 *  requirements apply also.
+		 */
+		struct {
+			/** Packet buffer pointers must be aligned to the buffer size.
+			  * For example, if the buffer size would be 2304 bytes (0x900),
+			  * each buffer start address must be a multiple of 0x900
+			  * (e.g. 0x12000900, 0x12001200, 0x12004800, etc). */
+			uint16_t buf_size_aligned : 1;
+		};
+
+		/** Maximum headroom parameter value
+		 *
+		 *  The packet pool headroom parameter may not exceed this value.
+		 */
+		uint32_t max_headroom;
+
+		/** Maximum headroom size in bytes
+		 *
+		 *  Any newly allocated packet will have at most this much headroom. Application
+		 *  may use this to ensure that packet buffer size is large enough to fit both
+		 *  buffer headers, headroom and data.
+		 */
+		uint32_t max_headroom_size;
+
+		/** Maximum number of segments per packet */
+		uint32_t max_segs_per_pkt;
+
+		/** Maximum user area size in bytes */
+		uint32_t max_uarea_size;
+
+	} pkt;
+
+} odp_pool_ext_capability_t;
+
+/**
+ * External memory pool parameters
+ */
+typedef struct odp_pool_ext_param_t {
```

Comment:
@psavol odp_pool_info_t has odp_pool_param_t

So what should be the behavior of odp_pool_info() API for external pools ?

Reply:
Added in v13.